uPortal-security/uPortal-security-permissions/src/main/java/org/apereo/portal/security/provider/AuthorizationImpl.java

```diff
@@ -241,7 +241,15 @@ public boolean canPrincipalConfigure(
         }
 
         final String activity = IPermission.PORTLET_MODE_CONFIG;
-        return doesPrincipalHavePermission(principal, owner, activity, target);
+
+        boolean isAllowed = doesPrincipalHavePermission(principal, owner, activity, target);
+        if (this.log.isTraceEnabled()) {
+            log.trace(
+                    String.format(
+                            "In canPrincipalConfigure() - principal.key=[%s], is allowed?=[%s]",
+                            principal.getKey(), isAllowed));
+        }
```

Comment:
The `org.slf4j.Logger` turns this mess into 1 line -- it handles both `isTraceEnabled` and `String.format` automatically (though the syntax is different).